io-sim/src/Control/Monad/IOSim.hs

```diff
@@ -735,31 +746,31 @@ execAtomically mytid = go [] []
   where
     go :: [SomeTVar s] -> [SomeTVar s] -> TVarId
        -> StmA s a -> ST s (StmTxResult s a)
-    go read written nextVid action = case action of
-      ReturnStm x     -> do (vids, tids) <- finaliseCommit written
+    go read_tvs written_tvs nextVid action = case action of
```

Comment:
What's all this renaming for? We can hide imports to avoid name clashes.

Reply:
I proposed this as it is quite ugly to have a variable named the same as some function in Prelude. And, in this case the name `read` is not very indicative.
The alternative is to import the Prelude's read function again in a qualified import statement. More ugliness!

Reply:
Fair enough :-)